eip-0042.md

```diff
@@ -0,0 +1,174 @@
+
+Then we will manually create the ErgoTree; since using the compiler could cause the resulting address to vary.
+
+The address derivation algorithm is as follows:
+
+Sort the list of signers' public keys before being used in the contract.
+Create the required ergoTree array as follows:
+
+- The first byte in the array is `10`.
+- The number of constant values in the contract is encoded as a variable-length quantity (VLQ).
+  This number equals `M+1` (Constant `N` and `M` public keys)
+- A byte with value `04` denotes the integer type,followed by the value of `N` encoded as VLQ.
+- For each public key 35 bytes are used: 2 bytes with value `08cd`, followed by 33 bytes representing the public key.
+- A byte with the value `98` indicates **"atLeast"**.
+- Two bytes, with values `73` and `00`, respectively.
+  The value `00` indicates the VLQ-encoded index of value `N` in the list of constants.
+- A constant byte with value `83`, the VLQ-encoded value of `M`, and a constant byte with value `08`
+- For each public key (`M` times), a byte with value `73`,followed by the VLQ-encoded value of the index of the public key in the list of constants
+
+Therefore, for each derivation path, all wallets are able to generate a unique and similar byte array as an ErgoTree.
+
+![Ergo Tree Bytes](./eip-0042/bytes.png)
+
+## Transaction Signing
+
+An unsigned transaction is generated on one wallet. This transaction may have been generated through ErgoPay or the dApp connector, or it could be an outgoing transaction to another address through the wallet itself.
+
+After an unsigned transaction is generated, the wallet
+
+Generates a commitment.
+Stores its private part locally.
+Shares a commitment as below
+
+```
+{
+  tx:<reduced transaction bytes encoded as base64>,
+  boxes:[encoded boxes as base64],
+  commitment: commitments encoded
+}
+```
+
+The input `boxes:` are only used to display transactions on wallets.
+
+The general commitment format is as follows. However, this general-purpose commitment can be used for any contract.
+```
+{
+    "hint": "cmtReal",
+    "position": "0-0",
+    "type": "dlog",
+    "pubkey": {
+      "op": "205",
+      "h": "03a73c66970f14fc6450c6ab1a167cb4ba3baa64b20f731e22ec6840c70d27ef1c",
+    }
+    "a": "0253db866791af521ba4ab009509b6db89d272d9461636ee65eaa3e316884b21a4"
+}
+```
+
+In a multi-signature wallet, a simplified version of commitment is used.
+
+```
+commitments: [
+  [<base64 string of commitment for public key index 0 in input 0>, ...]
+  [<base64 string of commitment for public key index 0 in input 1>, ...]
+]
+```
+
+In the multi-signature wallet, a commitment is represented as a list of base64 strings to reduce the commitment size. Also, an empty string is used wherever a commitment is unavailable.
+
+[//]: # (and we can transform these commitments to standard format:)
+This reduced format can be transformed into the standard format:
+
+```
+{
```

Comment:
This format differs from the actual format in [EIP-11](https://github.com/ergoplatform/eips/blob/9906f403f61d8592ab955494cc4310563f4067c6/eip-0011.md?plain=1#L104) (index/position, pubkey/pk, a/commitment). Is EIP-11 outdated?